#!/usr/bin/env python3

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "ros_rknn_yolo"

gen = ParameterGenerator()

gen.add("enable_streaming_detect", bool_t, 0, "whether to start steaming detection", False)
gen.add("enable_draw", bool_t, 0, "whether to publish img with result", False)
gen.add("enable_crop", bool_t, 0, "whether to publish msg with crop detected objects img", False)
gen.add("raw_img_topic_name", str_t, 0, "the steaming detection subscribe topic name", "/image_raw")
gen.add("drop_interval", int_t, 0, "perform detection every drop_interval frame interval", 1, 0, 30)

exit(gen.generate(PACKAGE, "ros_rknn_yolo", "RknnYolo"))

